File: 0242-valid-anagram/0242-valid-anagram.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        
        unordered_map<char, int> freq;

        for(int i=0;i<s.size();i++)
        {
            freq[s[i]]++;
        }

        for(int i=0;i<t.size(); i++)
        {
            freq[t[i]]--;

            if(freq[t[i]] == 0)
            {
                freq.erase(t[i]);
            }
        }

        return (freq.empty() ? true : false);

    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        mp = {}

        if len(s) != len(t):
            return False

        for i in s:
            if i in mp:
                mp[i] += 1
            else:
                mp[i] = 1
        
        for c in t:
            if c not in mp:
                return False
            else:
                mp[c] -= 1
                if mp[c] == 0:
                    mp.pop(c)

        if len(mp) > 0:
            return False

        return True
    